### DISCUSSION TITLE
NimBLE support for ESP32-C6, H2 and future MCU's and why is Bluedroid still the default?

### DISCUSSION THREAD

**Discussion by h2zero** [espressif/arduino-esp32#9836] [Ideas]
### Board

esp32-c6-devkitc-1

### Device Description

esp32-c6-devkitc-1

### Hardware Configuration

nothing

### Version

latest master (checkout manually)

### IDE Name

all

### Operating System

all

### Flash frequency

all

### PSRAM enabled

yes

### Upload speed

any

### Description

First, sorry for skirting the default fields, I'm opening this here for discussion and exposure.

Times have changed, classic Bluetooth has become, well, classic, yet the default Bluetooth/BLE stack is still Bluedroid, even though no other Espressif SOC except the original ESP32 has supported Bluetooth classic, why? 

This is detrimental to the users that are using one of the many other MCU's that are not the original ESP32.

The latest release has shined the spotlight on the issue, as of now the [NimBLE-Arduino](https://github.com/h2zero/NimBLE-Arduino) library is not able to support the C6 and H2 variants in the 3.x.x releases because the `libbt` has Bluedroid enabled and the controller is NimBLE based which causes all kinds of conflicts. Would it not make sense to enable NimBLE host for the "other-than-original-esp32-devices" and incorporate the supporting library [esp-nimble-cpp](https://github.com/h2zero/esp-nimble-cpp), which NimBLE-Arduino is derived from and many use?

Alternatively, maybe have an option to select the BLE stack and link a different 'libbt' for each stack?

If none of this is viable, I understand, close this at your leisure, just thought I'd open some discussion.

### Sketch

```cpp
#include "NimBLEDevice.h"

void setup(){}

void loop(){}
```

### Debug Message

```plain
Broke..
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by me-no-dev** [comment]
Moving this to discussion instead

**Comment by Jason2866** [comment]
Would be great if this would be done. I do exactly this for our Tasmota fork. Well not exactly the same since I enable NimBLE for all MCUs ;-) and use the h2zero esp-nimble-cpp for all BLE stuff. Don't need classic BT for esp32....  It just works :-)

**Reply by greubon** [reply]
Hello, how do you proceed to "enable NimBLE for all MCUs"?
I am currently trying to use esp-nimble-cpp for a H2 in ESP-IDF (with Aruidno as component), and I'm struggling to make it work.
It works with an ESP32 WROOM, but compilation fails when it turns to do it for a H2...

**Reply by Jason2866** [reply]
H2 is not tested with `esp-nimble-cpp` You have to add the H2 here https://github.com/h2zero/esp-nimble-cpp/blob/cabd48aef3c2c7b2947007f76e190633a5ce12fe/CMakeLists.txt#L28

**Reply by greubon** [reply]
Perfect, it compiles now. For some reasons also my IDF target in variable environment was not well defined too (done using $env:IDF_TARGET="esp32h2). So far the simple client example works, at least. Thank you very much!

**Comment by h2zero** [comment]
Nobody looks at discussions, I guess this is where it dies... 👎

**Reply by me-no-dev** [reply]
not true. And this is already in discussion internally also

**Reply by jhmaloney** [reply]
Can someone post a link to the discussion for this topic?

As an enthusiastic user of NimBLE-Arduino and the entire ESP32 family, I would love to see NimBLE as an option for all ESP32 boards that can support Bluetooth. FYI, I started with Bluedroid but I found that NimBLE had a smaller RAM footprint and excellent performance. (I am only using BLE, not Bluetooth Classic.)

**Reply by Jason2866** [reply]
@jhmaloney Here https://github.com/espressif/arduino-esp32/discussions/9836
You may try the Platformio build h2zero and I are working on. WiP but should already work
```
platform = https://github.com/jason2866/platform-espressif32.git#Arduino/IDF-NimBLE
```

**Reply by h2zero** [reply]
> not true. And this is already in discussion internally also

My apologies @me-no-dev, I was frustrated by spending many hours and not being able to work around the issues that make supporting the new chips impossible. Not many people look at discussions and your own team opened an issue [here](https://github.com/espressif/arduino-esp32/issues/9878) due to visibility I suppose. I have commented there and what I have said would also allow me to provide support for the new chips.

**Reply by me-no-dev** [reply]
Switching the BLE stack is no easy decision. The BLE lib has been around now for some good time and we do expect some people to get angry if we switch, so we need to do it properly and easy. Decision has been made and this will happen, but I can not give a timeframe yet. We are a small team and have many things to take care of

**Reply by jhmaloney** [reply]
FWIW, MicroBlocks started out with the Espressif Bluetooth stack. When we switched to NimBLE, the Flash and RAM footprints went down and performance was the same or better. Of course, nobody likes to be forced to change code that is already written so changing BLE stacks would certainly cause some pain but there are benefits as well.

**Comment by Swiftnesses** [comment]


**Reply by Jason2866** [reply]
What is a bad decision, that the change to NimBLE is done taking care to do it nice?
